packages/iov-lisk/src/liskclient.ts

```diff
@@ -0,0 +1,200 @@
+  Address,
+  BcpAccount,
+  BcpAccountQuery,
+  BcpAddressQuery,
+  BcpNonce,
+  BcpQueryEnvelope,
+  BcpTicker,
+  BcpTransactionResponse,
+  ConfirmedTransaction,
+  IovReader,
+  Nonce,
+  TokenTicker,
+} from "@iov/bcp-types";
+import { Encoding } from "@iov/encoding";
+import { Algorithm, ChainId, PostableBytes, PublicKeyBytes, Tag, TxId, TxQuery } from "@iov/tendermint-types";
+
+import { Parse } from "./parse";
+
+function isAddressQuery(query: BcpAccountQuery): query is BcpAddressQuery {
+  return (query as BcpAddressQuery).address !== undefined;
+}
+
+/**
+ * Encodes the current date and time as a nonce
+ */
+export function generateNonce(): Nonce {
+  const now = new ReadonlyDate(ReadonlyDate.now());
+  return Parse.timeToNonce(now);
+}
+
+export class LiskClient implements IovReader {
+  private readonly baseUrl: string;
+
+  constructor(baseUrl: string) {
+    this.baseUrl = baseUrl;
+  }
+
+  public disconnect(): void {
+    // no-op
+  }
+
+  public async chainId(): Promise<ChainId> {
+    const url = this.baseUrl + "/api/node/constants";
+    const result = await axios.get(url);
+    const responseBody = result.data;
+    return responseBody.data.nethash;
+  }
+
+  public async height(): Promise<number> {
+    const url = this.baseUrl + "/api/node/status";
+    const result = await axios.get(url);
+    const responseBody = result.data;
+    return responseBody.data.height;
+  }
+
+  public async postTx(bytes: PostableBytes): Promise<BcpTransactionResponse> {
+    const transactionId = JSON.parse(Encoding.fromUtf8(bytes)).id as string;
+    if (!transactionId.match(/^[0-9]+$/)) {
+      throw new Error("Invalid transaction ID");
+    }
+
+    await axios.post(this.baseUrl + "/api/transactions", bytes, {
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+
+    // Sleep some seconds to ensure transaction will be found.
+    // Sleep duration determined by trial and error. 15 seconds was not enough.
+    await new Promise(resolve => setTimeout(resolve, 20 * 1000));
+
+    const result = await axios.get(this.baseUrl + `/api/transactions?id=${transactionId}`);
+    const responseBody = result.data;
+
+    let height: number | undefined;
+    let transactionResultBytes: Uint8Array | undefined;
+    if (responseBody.meta.count === 1) {
+      const transactionResult = responseBody.data[0];
+      height = transactionResult.height;
+      transactionResultBytes = Encoding.toUtf8(JSON.stringify(transactionResult));
+    }
+
+    return {
+      metadata: {
+        status: true, // commit failures should throw
+        height: height,
+      },
+      data: {
+        message: "",
+        txid: Encoding.toAscii(transactionId) as TxId,
+        result: transactionResultBytes || new Uint8Array([]),
+      },
+    };
+  }
+
+  public getTicker(_: TokenTicker): Promise<BcpQueryEnvelope<BcpTicker>> {
+    throw new Error("Not implemented");
+  }
+
+  public getAllTickers(): Promise<BcpQueryEnvelope<BcpTicker>> {
```

Comment:
Can be hardcoded to just return LSK?
But we should talk to isabella first on the design there.
So we can differentiate testnet from mainnet

Reply:
hmmm... i guess this codec also needs to support lisk-compatible forks as well...
if the ticker is not in the api anywhere, maybe we hardcode a chainId->ticker mapping.
maybe later read that mapping in from bns